src/libraries/System.Formats.Tar/tests/TarEntry/TarEntryGnu.Tests.cs

```diff
@@ -38,6 +38,218 @@ public void Constructor_UnsupportedEntryTypes()
+            {
+                Gid = TestGid,
+                GroupName = TestGName,
+                Uid = TestUid,
+                UserName = TestUName,
+            };
+
+            Assert.True(firstEntry.AccessTime > firstNow);
+            Assert.True(firstEntry.ChangeTime > firstNow);
+
+            DateTimeOffset secondNow = DateTimeOffset.Now;
+
+            V7TarEntry otherEntry = new V7TarEntry(other: firstEntry);
+            Assert.Equal(TarEntryType.V7RegularFile, otherEntry.EntryType);
+
+            GnuTarEntry secondEntry = new GnuTarEntry(other: otherEntry);
+            Assert.Equal(TarEntryType.RegularFile, secondEntry.EntryType);
+
+            Assert.True(secondEntry.AccessTime > secondNow);
+            Assert.True(secondEntry.ChangeTime > secondNow);
+            Assert.Equal(TestGid, secondEntry.Gid);
+            Assert.Equal(DefaultGName, secondEntry.GroupName);
+            Assert.Equal(TestUid, secondEntry.Uid);
+            Assert.Equal(DefaultUName, secondEntry.UserName);
+        }
+
+        [Fact]
+        public void Constructor_ConversionUstar_BackAndForth()
+        {
+            DateTimeOffset firstNow = DateTimeOffset.Now;
+            GnuTarEntry firstEntry = new GnuTarEntry(TarEntryType.BlockDevice, "blockdev")
+            {
+                DeviceMajor = TestBlockDeviceMajor,
+                DeviceMinor = TestBlockDeviceMinor,
+                Gid = TestGid,
+                GroupName = TestGName,
+                Uid = TestUid,
+                UserName = TestUName,
+            };
+
+            Assert.True(firstEntry.AccessTime > firstNow);
+            Assert.True(firstEntry.ChangeTime > firstNow);
+
+            DateTimeOffset secondNow = DateTimeOffset.Now;
+
+            UstarTarEntry otherEntry = new UstarTarEntry(other: firstEntry);
+
+            GnuTarEntry secondEntry = new GnuTarEntry(other: otherEntry);
+
+            Assert.True(secondEntry.AccessTime > secondNow);
+            Assert.True(secondEntry.ChangeTime > secondNow);
+            Assert.Equal(TestBlockDeviceMajor, secondEntry.DeviceMajor);
+            Assert.Equal(TestBlockDeviceMinor, secondEntry.DeviceMinor);
+            Assert.Equal(TestGid, secondEntry.Gid);
+            Assert.Equal(TestGName, secondEntry.GroupName);
+            Assert.Equal(TestUid, secondEntry.Uid);
+            Assert.Equal(TestUName, secondEntry.UserName);
+        }
+
+        [Fact]
+        public void Constructor_ConversionPax_BackAndForth()
+        {
+            DateTimeOffset firstNow = DateTimeOffset.Now;
+            GnuTarEntry firstEntry = new GnuTarEntry(TarEntryType.BlockDevice, "blockdev")
+            {
+                DeviceMajor = TestBlockDeviceMajor,
+                DeviceMinor = TestBlockDeviceMinor,
+                Gid = TestGid,
+                GroupName = TestGName,
+                Uid = TestUid,
+                UserName = TestUName,
+            };
+
+            Assert.True(firstEntry.AccessTime > firstNow);
```

Comment:
This is intermittently failing in this PR's CI: I am trying to verify that the initial constructor sets the AccessTime and ChangeTime to DateTimeOffset.Now, which should be a value larger than `firstNow`.

Reply:
In the latest commits, I am now creating `firstNow` with a substraction of 100 milliseconds. I am suspecting the timestamp is too close to the one used in the constructor and maybe the comparison is not passing due to how close they both are.

I also moved the `secondNow` timestamp to be created before _both_ conversions, so that the extra work adds some time difference before the actual comparison is done.